apps/dashboard-service/src/dashboard/dto/cache.service.ts

```diff
@@ -54,14 +54,18 @@ export class CacheService {
   }
 
   getCacheParameterForProgrammes(programmes: string[]) {
-    if (programmes == null || programmes.length === 0) {
-      return "";
+    if (typeof programmes === "string") {
+      return (programmes = [programmes]);
     }
-    const sortedProgrammes = programmes.sort();
-    return sortedProgrammes.join(",");
+
+    return programmes.length === 0 ? "" : programmes.sort().join(",");
```

Comment:
A couple of things here:
* :art: You can leave as-is, but I generally avoid `typeof` and would use `if (!Array.isArray(programmes))` above
* There's no reason to assign to `programmes` when returning. That line can simply be `return [programmes];`
* Why is the if block above returning an array, but this line is intentionally converting an array to a string? 
* Same questions for the similar methods below.